ratex/core/distributed/ratex_fully_sharded_data_parallel.py

```diff
@@ -0,0 +1,85 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Module wrapper implementing ZeRO-1 in an FSDP style interface"""
+import torch.nn as nn
+import torch.nn.functional as F
+
+from raf import distributed as dist
+
+import ratex.optimizer as optim
+from ratex.core.lazy_model import all_gather, all_reduce, REDUCE_SUM
+
+
+class RatexFullyShardedDataParallel(nn.Module):
+    """FSDP ZeRO-1 wrapper"""
+
+    def __init__(self, module, optimizer=optim.SGD, optimizer_config=None):
```

Comment:
1. Add docstring with argument types.
2. Add arguments types to all member functions.
Ref: https://github.com/awslabs/ratex/blob/main/ratex/core/lazy_model.py

Reply:
I added arguments as needed according to our convention.